### PR TITLE
DEX-817 Incorrect Waves balance after a WebSocket connection

### PR DESCRIPTION
**Pull request by vsuharnikov**

### CODE REVIEW

File: waves-ext/src/main/scala/com/wavesplatform/dex/grpc/integration/services/WavesBlockchainApiGrpcService.scala [hunk 1]
```diff
@@ -260,22 +259,16 @@ class WavesBlockchainApiGrpcService(context: ExtensionContext, balanceChangesBat
...
 
-      val address              = request.address.toVanillaAddress
-      val pessimisticPortfolio = context.blockchain.portfolio(address) |+| context.utx.pessimisticPortfolio(address)
-
-      val pessimisticPortfolioNonZeroBalances = {
-        if (pessimisticPortfolio.balance == 0) pessimisticPortfolio.assets
-        else pessimisticPortfolio.assets ++ Map(Waves -> pessimisticPortfolio.balance)
+      val finalPortfolio = AllAssetsSpendableBalanceResponse.Record(Waves.toPB, context.utx.spendableBalance(address, Waves)) ::
+        portfolio.assets.toList.map {
+        case (asset, _) => AllAssetsSpendableBalanceResponse.Record(asset.toPB, context.utx.spendableBalance(address, asset))
```
Comment (DenisSerebryansky) [inline waves-ext/src/main/scala/com/wavesplatform/dex/grpc/integration/services/WavesBlockchainApiGrpcService.scala:268]:
```
AllAssetsSpendableBalanceResponse(
  (Waves :: pessimisticPortfolio.assets.keys.toList)
    .map(a => AllAssetsSpendableBalanceResponse.Record(a.toPB, context.utx.spendableBalance(address, a)))
    .filterNot(_.balance == 0L)
)
```

File: dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsAddressStreamTestSuite.scala [hunk 2]
```diff
@@ -416,5 +417,27 @@ class WsAddressStreamTestSuite extends WsSuiteBase with TableDrivenPropertyCheck
...
+      broadcastAndAwait(leaseTx)
+      dex1.start()
+
+      val wsc = mkDexWsConnection(dex1)
+      wsc.send(WsAddressSubscribe(bob, WsAddressSubscribe.defaultAuthType, mkJwt(bob)))
+
+      eventually {
+        val balance = wsc.receiveAtLeastN[WsAddressState](1).map(_.balances).squashed - btc - wct
+        balance should matchTo(
+          Map[Asset, WsBalances](
```
Comment (DenisSerebryansky) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsAddressStreamTestSuite.scala:435]: Maybe we can automate this widespread cast somehow?

Comment (vsuharnikov) [reply dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsAddressStreamTestSuite.scala:435]: Probably by `def mkBalanceChanges(xs: (Assets, Long)*)`. See no reason to do this, it's the only case.

File: waves-ext/src/main/scala/com/wavesplatform/dex/grpc/integration/services/WavesBlockchainApiGrpcService.scala (hunk 1 above)
Comment (vsuharnikov) [reply waves-ext/src/main/scala/com/wavesplatform/dex/grpc/integration/services/WavesBlockchainApiGrpcService.scala]: Fixed

**Pull request merged**